espnet/nets/pytorch_backend/e2e_tts.py

```diff
@@ -235,10 +171,96 @@ class Tacotron2(torch.nn.Module):
+                            help='Number of postnet channels')
+        parser.add_argument('--postnet_filts', default=5, type=int,
+                            help='Filter size of postnet')
+        parser.add_argument('--output_activation', default=None, type=str, nargs='?',
+                            help='Output activation function')
+        # cbhg
+        parser.add_argument('--use_cbhg', default=False, type=strtobool,
+                            help='Whether to use CBHG module')
+        parser.add_argument('--cbhg_conv_bank_layers', default=8, type=int,
+                            help='Number of convoluional bank layers in CBHG')
+        parser.add_argument('--cbhg_conv_bank_chans', default=128, type=int,
+                            help='Number of convoluional bank channles in CBHG')
+        parser.add_argument('--cbhg_conv_proj_filts', default=3, type=int,
+                            help='Filter size of convoluional projection layer in CBHG')
+        parser.add_argument('--cbhg_conv_proj_chans', default=256, type=int,
+                            help='Number of convoluional projection channels in CBHG')
+        parser.add_argument('--cbhg_highway_layers', default=4, type=int,
+                            help='Number of highway layers in CBHG')
+        parser.add_argument('--cbhg_highway_units', default=128, type=int,
+                            help='Number of highway units in CBHG')
+        parser.add_argument('--cbhg_gru_units', default=256, type=int,
+                            help='Number of GRU units in CBHG')
+        # model (parameter) related
+        parser.add_argument('--use_batch_norm', default=True, type=strtobool,
+                            help='Whether to use batch normalization')
+        parser.add_argument('--use_concate', default=True, type=strtobool,
+                            help='Whether to concatenate encoder embedding with decoder outputs')
+        parser.add_argument('--use_residual', default=True, type=strtobool,
+                            help='Whether to use residual connection in conv layer')
+        parser.add_argument('--dropout', default=0.5, type=float,
+                            help='Dropout rate')
+        parser.add_argument('--zoneout', default=0.1, type=float,
+                            help='Zoneout rate')
+        parser.add_argument('--reduction_factor', default=1, type=int,
+                            help='Reduction factor')
+        # loss related
+        parser.add_argument('--use_masking', default=False, type=strtobool,
+                            help='Whether to use masking in calculation of loss')
+        parser.add_argument('--bce_pos_weight', default=20.0, type=float,
+                            help='Positive sample weight in BCE calculation (only for use_masking=True)')
+        return
+
     def __init__(self, idim, odim, args):
+        # initialize base classes
         super(Tacotron2, self).__init__()
```

Comment:
Use `TTSInterface.__init__(self)`  because C4 algorithm is not popular for many python user.  Namely, I am not sure the `super` is `TTSInterface.__init__` or `nn.Module.__init__`.

Reply:
Actually, I’ve never use multiple inheritance so which way is correct to initialize.
I followed your manner. Thanks.